contracts/Orchestrator.sol

```diff
@@ -46,35 +48,107 @@ contract Orchestrator is Ownable {
     {
         require(msg.sender == tx.origin);  // solhint-disable-line avoid-tx-origin
 
+        // call monetary policy rebase, always revert on failure
         policy.rebase();
 
-        for (uint i = 0; i < transactions.length; i++) {
-            Transaction storage t = transactions[i];
-            if (t.enabled) {
-                bool result =
-                    externalCall(t.destination, t.data);
-                if (!result) {
-                    emit TransactionFailed(t.destination, i, t.data);
-                    revert("Transaction Failed");
-                }
-            }
+        // call peripheral contracts, handle reverts based on policy
+        for (uint index = 0; index < transactions.length; index++) {
+            _executePeripheralTransaction(index);
         }
     }
 
+    /**
+     * @notice Get the revert message and code from a call.
+     * @param index uint256 Index of the transaction.
+     * @return revertMessage string Revert message.
+     * @return revertCode bytes32 Revert code.
+     */
+    function _executePeripheralTransaction(uint256 index) internal returns (bool, bytes[] memory) {
+        // declare storage reference
+        Transaction storage transaction = transactions[index];
+
+        // validate sufficient gas left
+        require(gasleft() > transaction.gasLimit);
+
+        // perform external call
+        // todo: @thegostep solc v0.4 does not return revert string on call
+        // todo: @thegostep solc v0.4 does not support specifying gaslimit on call
+        // decide if upgrade solc or implement in assembly
+        // https://solidity.readthedocs.io/en/v0.4.24/units-and-global-variables.html#address-related
+        (bool success, bytes memory res) = address(transaction.destination).call(transaction.data);
```

Comment:
check if can upgrade solc version to latest ToB version without breaking everything
if requires re-writing test suite, consider assembly